.github/workflows/linelint.yml

```diff
@@ -0,0 +1,27 @@
+name: EOF Newline Check
+
+on: [push, pull_request]
+
+permissions:
+  contents: read
+
+jobs:
+  eof-check:
+    runs-on: ubuntu-latest
+    name: Check for missing EOF newlines
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Get changed files
+        id: changed-files
+        run: |
+          FILES=$(git diff --name-only --diff-filter=d origin/master...HEAD | tr '\n' ' ')
```

Comment:
I don't think this is correct, since not every PR is submitted to the `master` branch.